pkg/agent/sysadvisor/plugin/qosaware/server/server.go

```diff
@@ -43,6 +43,8 @@ type subQRMServer interface {
 	Name() string
 	Start() error
 	Stop() error
+	RegisterAdvisorServer()
+	UpdateContainerResources(request *advisorsvc.AddContainerRequest, containerInfo *types.ContainerInfo)
```

Comment:
if so, I prefer to add an `AddContainer` function in subQRMServer rather than `UpdateContainerResources` (it seems too specific) and you can firstly call `AddContainer` in subQRMServer, and then assemble extra and additional info in base-server

Reply:
also, please add some comments here to explain the functionality of the added functions

Reply:
Now reflect mechanism is used to find corresponding filed of ContainerInfo